Makefile

```diff
@@ -154,7 +154,7 @@ index-rollover-integration-test: docker-images-elastic
 
 .PHONY: cover
 cover: nocover
-	$(GOTEST) -tags=memory_storage_integration -timeout 5m -coverprofile cover.out ./...
+	$(GOTEST) -tags=memory_storage_integration -timeout 5m -coverprofile cover.out ./... > test-results.json
```

Comment:
@albertteoh what makes it a JSON file? If I change it to `| tee test-results.json` then I see the standard text output:
```
=== RUN   TestRegisterStaticHandler/basePath=/
=== RUN   TestRegisterStaticHandler/basePath=/jaeger
--- PASS: TestRegisterStaticHandler (0.03s)
    --- PASS: TestRegisterStaticHandler/basePath= (0.01s)
```
If it is meant to be std output, then I would suggest changing to `|tee` because it makes the CI logs show the test progress.

Reply:
The `test-ci` make target defines `GOTEST` as: `GOTEST := $(GOTEST_QUIET) -json`; so it's the extra `-json` flag that makes the output json: https://github.com/jaegertracing/jaeger/blob/main/Makefile#L460.

So the full command would resemble (with `| tee`):
```shell
go test -race -json -tags=memory_storage_integration -timeout 5m -coverprofile cover.out ./... | tee test-results.json
```

Adding a `| jq .` to the end of that, and the output looks like:
```json
{
  "Time": "2023-12-27T21:40:21.085452+11:00",
  "Action": "pass",
  "Package": "github.com/jaegertracing/jaeger/storage/spanstore/metrics",
  "Elapsed": 3.848
}
```

It's a good callout though, because I hadn't considered the impact of this change on the stdout display of test results which are useful for debugging unit test runs. Maybe test summaries are not worth the trade-off for harder to read test output?

What do you think?

Reply:
I see. I think `jq` will be overkill, but changing to `tee` is good. The `test-ci` is not really meant to be run locally, but with `-json` flag the output is still readable
```
{"Time":"2023-12-27T10:16:06.335612-05:00","Action":"start","Package":"github.com/jaegertracing/jaeger"}
{"Time":"2023-12-27T10:16:06.72618-05:00","Action":"run","Package":"github.com/jaegertracing/jaeger","Test":"TestDummy"}
```
(btw `jq` wouldn't work at all because ^ is not a well-formed JSON, it has no top-level container).

Reply:
Cool, I've created a PR to use ` | tee`: #5050.

> (btw jq wouldn't work at all because ^ is not a well-formed JSON, it has no top-level container)

Running it locally, `jq` seemed to be okay with it; it just outputs separate JSON blobs:
```shell
go test -race -json -tags=memory_storage_integration -timeout 5m -coverprofile cover.out ./... | tee test-results.json | jq .
```

```json
{
  "Time": "2023-12-28T07:02:35.584992+11:00",
  "Action": "output",
  "Package": "github.com/jaegertracing/jaeger",
  "Test": "TestDummy",
  "Output": "=== RUN   TestDummy\n"
}
{
  "Time": "2023-12-28T07:02:35.585097+11:00",
  "Action": "output",
  "Package": "github.com/jaegertracing/jaeger",
  "Test": "TestDummy",
  "Output": "--- PASS: TestDummy (0.00s)\n"
}
```

Reply:
there's also the question of pipe buffering, I think it gets worse so we'd less the output less frequently (but maybe I was just impatient). Anyway, I think one line is fine for now. I am more concerned with now failing the rest of the workflow when test results cannot be uploaded for whatever reason.

Reply:
> I am more concerned with now failing the rest of the workflow when test results cannot be uploaded for whatever reason.

There are no upload/download steps in our test summary reporting workflows at the moment. You're right though, because I vaguely recall codecov upload failures may have failed the rest of the workflow in the past.

If we do want test summary reports on our PRs as comments and in the check runs, then artifact uploads and downloads would be required. Is this feature desirable enough to warrant introducing uploads? If not, I can hit the pause button on my WIP.

Reply:
I don't feel like it's worth the hustle. The fact that test results are already showing in the summary is good, although if all it does is show the count (not the details of the failures) then it's pretty useless. Adding comments and checks is nice to have, but not at the expense of stability of the CI, which has been happening in the past few days due to these changes.